src/main/java/com/cema/activity/controllers/WeighingController.java

```diff
@@ -0,0 +1,157 @@
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import io.swagger.annotations.ResponseHeader;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+@RequestMapping("/v1")
+@Api(produces = "application/json", value = "Allows interaction with the weighing database. V1")
+@Validated
+@Slf4j
+public class WeighingController {
+
+    private static final String BASE_URL = "/weightings";
+    private static final String HANDLER_TYPE = "weighing";
+    private final ActivityHelper activityHelper;
+
+    public WeighingController(ActivityHelper activityHelper) {
+        this.activityHelper = activityHelper;
+    }
+
+    @ApiOperation(value = "Retrieve a weighing by weighing id")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully found Weighing"),
+            @ApiResponse(code = 404, message = "Weighing not found")
+    })
+    @GetMapping(value = BASE_URL + "/{activity_id}", produces = {MediaType.APPLICATION_JSON_VALUE})
+    public ResponseEntity<Weighing> getWeighingById(
+            @ApiParam(
+                    value = "The id of the weighing you are looking for.",
+                    example = "123")
+            @PathVariable("activity_id") String activityId,
+            @ApiParam(
+                    value = "The cuig of the establishment of the weighing. Only used if you are admin.",
+                    example = "312")
+            @RequestParam(value = "cuig", required = false) String cuig) {
+        log.info("Request for get Weighing received.");
+
+        Weighing result = (Weighing) activityHelper.get(activityId, HANDLER_TYPE, cuig);
+
+        return new ResponseEntity<>(result, HttpStatus.OK);
+    }
+
+    @ApiOperation(value = "Register a new weighing to the database")
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "Weighing created successfully"),
+            @ApiResponse(code = 401, message = "You are not allowed to register this weighing"),
```

Comment:
I think this should be 403, might be wrong

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403

Reply:
avoided 403 to reduce complexity.